src/evaluation_system/tests/plugin_manager_test.py

```diff
@@ -319,6 +319,45 @@ def test_get_history(dummy_settings, temp_user):
     assert g1[0] == g2[0]
 
 
+def test_get_history_all(dummy_settings, temp_user):
+    import random
+    import re
+    import socket
+    import string
+
+    from django.contrib.auth.models import User
+
+    import evaluation_system.api.plugin_manager as pm
+    from evaluation_system.model.history.models import History
+
+    pm.run_tool("dummyplugin", user=temp_user)
+
+    N = 10
+    random_string = "".join(random.choices(string.ascii_lowercase + string.digits, k=N))
+    other_user = User.objects.create_user(
+        username=f"user_{random_string}", password="123"
+    )
+    History.objects.create(
+        timestamp=datetime.datetime.now(),
+        status=History.processStatus.running,
+        uid=other_user,
+        configuration='{"some": "config", "dict": "values"}',
+        tool=f"dummytool_{random_string}",
+        slurm_output="/path/to/slurm-44742.out",
+        host=socket.gethostbyname(socket.gethostname()),
+    )
+    other_user.delete()
+
+    res1 = pm.get_history(user=temp_user)
+    res2 = pm.get_history_all(user=temp_user)
```

Comment:
Alright, I think we cannot use the simple `==` to compare two querysets `res1` and `res2` as they will always be unequal due to their underlying implementation. so, i suggest the following. I have already run this on my machine and it `PASSED`:

```python
def test_get_history_all(dummy_settings, temp_user):
    import random
    import re
    import socket
    import string
    import unittest
    class DjangoAssertQuerysetEqual(unittest.TestCase):
        pass
    django_equal_test_case = DjangoAssertQuerysetEqual()

    from django.contrib.auth.models import User

    import evaluation_system.api.plugin_manager as pm
    from evaluation_system.model.history.models import History

    pm.run_tool("dummyplugin", user=temp_user)

    N = 10
    random_string = "".join(random.choices(string.ascii_lowercase + string.digits, k=N))
    other_user = User.objects.create_user(
        username=f"user_{random_string}", password="123"
    )
    History.objects.create(
        timestamp=datetime.datetime.now(),
        status=History.processStatus.running,
        uid=other_user,
        configuration='{"some": "config", "dict": "values"}',
        tool=f"dummytool_{random_string}",
        slurm_output="/path/to/slurm-44742.out",
        host=socket.gethostbyname(socket.gethostname()),
    )
    other_user.delete()

    res1 = pm.get_history(user=temp_user)
    # just in case we want to use the second approach
    res_temp = pm.get_history(user=temp_user, get_all_users=True) 
    res2 = pm.get_history_all(user=temp_user)
    
    search_string1 = bool(re.search(f"dummytool_{random_string}", res1.__str__()))
    search_string2 = bool(re.search(f"dummytool_{random_string}", res2.__str__()))
    # to check if the two approaches have the same res
    django_equal_test_case.assertEqual(list(res2), list(res_temp)) 
    django_equal_test_case.assertNotEqual(list(res1), list(res2))
    assert search_string1 is False
    assert search_string2 is True
```

Reply:
Thanks!
I see what you mean, there are two diff queries and hence two diff objects. 
I realised that `assert set(res1)!=set(res2)` does actually what intended 

just pushed the changes

Reply:
yes u r totally right, in this case it's a good idea to use `set` instead of `assertEqual`. because we don't need to compare the content and in consequence take care of orders. 
Another idea would be just get the length, because in this test we are going to see adding `get_all_users=True` flag can add one more History objects than `False`

Reply:
that is also true.
I did that check with the search_string assert. But maybe with the length
assert len(res2) > len(res1)
should ahave been enougth